src/test/java/com/snowflake/kafka/connector/internal/streaming/SnowflakeSinkServiceV2IT.java

```diff
@@ -579,6 +581,89 @@ public void testNativeAvroInputIngestion() throws Exception {
+    // avro
+    SchemaBuilder schemaBuilder =
+        SchemaBuilder.struct()
+            .field("id", Schema.INT32_SCHEMA)
+            .field("first_name", Schema.STRING_SCHEMA)
+            .field("rating", Schema.FLOAT32_SCHEMA)
+            .field("approval", Schema.BOOLEAN_SCHEMA)
+            .field(
+                "info_map", SchemaBuilder.map(Schema.STRING_SCHEMA, Schema.INT32_SCHEMA).build());
+    Struct original =
+        new Struct(schemaBuilder.build())
+            .put("id", 42)
+            .put("first_name", "zekai")
+            .put("rating", 0.99f)
+            .put("approval", true)
+            .put("info_map", Collections.singletonMap("field", 3));
+
+    String schemaString =
+        "{\n"
+            + "            \"type\":\"record\",\n"
+            + "            \"name\":\"value_schema\",\n"
+            + "            \"fields\":[\n"
+            + "                {\"name\":\"id\",\"type\":\"int\"},\n"
+            + "                {\"name\":\"first_name\",\"type\":\"string\"},\n"
+            + "                {\"name\":\"rating\",\"type\":\"float\"},\n"
+            + "                {\"name\":\"approval\",\"type\":\"boolean\"},\n"
+            + "               "
+            + " {\"name\":\"info_map\",\"type\":{\"type\":\"map\",\"values\":\"string\"}}\n"
+            + "            ]\n"
+            + "        }";
+
+    SchemaRegistryClient schemaRegistry = new MockSchemaRegistryClient();
+    AvroConverter avroConverter = new AvroConverter(schemaRegistry);
+    avroConverter.configure(
+        Collections.singletonMap("schema.registry.url", "http://fake-url"), false);
+    byte[] converted = avroConverter.fromConnectData(topic, original.schema(), original);
+    ParsedSchema schema = new AvroSchema(schemaString);
+    schemaRegistry.register(topic + "-value", schema);
+    Map<String, String> schemaMap =
+        Utils.getSchemaFromSchemaRegistryClient(topic, schemaRegistry, "value");
+    conn.createTableWithSchema(table, schemaMap);
+
+    SchemaAndValue avroInputValue = avroConverter.toConnectData(topic, converted);
+
+    long startOffset = 0;
+    long endOffset = 0;
+
+    SinkRecord avroRecordValue =
+        new SinkRecord(
+            topic,
+            partition,
+            Schema.STRING_SCHEMA,
+            "test",
+            avroInputValue.schema(),
+            avroInputValue.value(),
+            startOffset);
+
+    SnowflakeSinkService service =
+        SnowflakeSinkServiceFactory.builder(conn, IngestionMethodConfig.SNOWPIPE_STREAMING, config)
+            .setRecordNumber(1)
+            .setErrorReporter(new InMemoryKafkaRecordErrorReporter())
+            .setSinkTaskContext(new InMemorySinkTaskContext(Collections.singleton(topicPartition)))
+            .addTask(table, new TopicPartition(topic, partition))
+            .build();
+    // Table should be created at the start of the task
+
+    service.insert(avroRecordValue);
```

Comment:
I would even go one more step and assert the columns returned by desc table/topic_name and make this a complete IT test. 

I think it is very important to also do a select * from the table and assert values from each column with what you expect it to be. 

Please add it in TestUtil.java class